### PR TITLE
Add chat display during multiplayer gameplay

### LINKED ISSUE #12402: Make multiplayer chat visible during gameplay / results
**Describe the new feature:** 

In osu!stable, when you are in a multiplayer room, the chat overlay is always opened and the multiplayer chat tab is fixed at the beggining of all the tabs.
In osu!lazer, the multiplayer chat is separate from the overlay which contain all the chat tabs, so it looks like there is no need to make the room's chat tab be fixed in the chat overlay, but when the game starts, it's common for players to use the room's tab in the chat overlay to communicate something to the players while you are in a rest section of the map or when the gameplay is loading at the beggining.
Currently, it's hard to find the current room's tab in osu!lazer because it is always at the end and can be confusing because the name is always "lazermp_" along with some numbers, the idea is to make the current room chat tab be fixed at the beggining of the tab list in the chat overlay and maybe rename it to "multiplayer" as stable does.

### PR DESCRIPTION
**Pull request by peppy**
For now, this toggles along with the other elements (based on HUD visibility). Seems to work okay for now.

Chat can also be focused via the `Tab` key.

- [x] Depends on https://github.com/ppy/osu/pull/14358.
- Closes https://github.com/ppy/osu/issues/12402

https://user-images.githubusercontent.com/191335/129681351-fd39af4e-2c15-4c4d-806b-f33d88520ba0.mp4

### CODE REVIEW

**Comment by smoogipoo** [comment]
Two things:
1. I don't think the textbox should ever focus during gameplay.
2. There's no way to select the textbox when you have overlay visibility mode set to "never". I'd expect pressing tab to show the overlay for the entire duration that the textbox is focused.

**Comment by peppy** [comment]
Hmm, so you want it to be non-interactive during gameplay? Maybe with the exception of the key being pressed?

**Comment by smoogipoo** [comment]
Yeah I can agree with non interactive unless keybind is pressed.

File: osu.Game.Tests/Visual/Multiplayer/TestSceneGameplayChatDisplay.cs [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+        [Test]
+        public void TestFocusOnTabKeyWhenExpanded()
+        {
+            assertChatFocused(false);
+            AddStep("press tab", () => InputManager.Key(Key.Tab));
+            assertChatFocused(true);
+        }
+
+        [Test]
+        public void TestFocusOnTabKeyWhenNotExpanded()
```
Comment (bdach) [inline osu.Game.Tests/Visual/Multiplayer/TestSceneGameplayChatDisplay.cs:94]: Judging from the code, this needs a `setLocalUserPlaying(true)` at the start to pass.

**Review by bdach — CHANGES_REQUESTED** [review]

**Comment by bdach** [comment]
I have one more gripe with this, namely that if I am able to manually enable the chat while playing using <kbd>Tab</kbd>, I'd expect to be able to hide it back again with the same keybind. Especially so that sometimes it may be pressed because of a misclick.

The only option available to dismiss the focused textbox as-is seems to be to click away from it with the mouse. Otherwise it keeps collecting keyboard input (even the gameplay Z/X inputs).

Something like this seems to work like I'd expect it to:

```diff
diff --git a/osu.Game/Screens/OnlinePlay/Multiplayer/GameplayChatDisplay.cs b/osu.Game/Screens/OnlinePlay/Multiplayer/GameplayChatDisplay.cs
index 5d494379e6..c56c8eedd5 100644
--- a/osu.Game/Screens/OnlinePlay/Multiplayer/GameplayChatDisplay.cs
+++ b/osu.Game/Screens/OnlinePlay/Multiplayer/GameplayChatDisplay.cs
@@ -72,10 +72,19 @@ public bool OnPressed(GlobalAction action)
             switch (action)
             {
                 case GlobalAction.FocusChatInput:
-                    expandedFromTextboxFocus.Value = true;
+                    if (!expandedFromTextboxFocus.Value)
+                    {
+                        expandedFromTextboxFocus.Value = true;
+
+                        // schedule required to ensure the textbox has become present from above bindable update.
+                        Schedule(() => Textbox.TakeFocus());
+                    }
+                    else
+                    {
+                        expandedFromTextboxFocus.Value = false;
+                        GetContainingInputManager().ChangeFocus(null);
+                    }
 
-                    // schedule required to ensure the textbox has become present from above bindable update.
-                    Schedule(() => Textbox.TakeFocus());
                     return true;
             }
 ```

File: osu.Game.Tests/Visual/Multiplayer/TestSceneGameplayChatDisplay.cs (hunk 1 above)
Comment (peppy) [reply osu.Game.Tests/Visual/Multiplayer/TestSceneGameplayChatDisplay.cs:94]: Weird, I guess I was always getting this from a previous step when testing.

**Review by bdach — APPROVED** [review]
code looks fine, going to ask for a double check wrt behaviour

**Review by smoogipoo — APPROVED** [review]

**Pull request merged**